model.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"github.com/taskcluster/taskcluster-client-go/queue"
+	"time"
+)
+
+// Used for modelling the xml we get back from Azure
+type QueueMessagesList struct {
+	XMLName       xml.Name       `xml:"QueueMessagesList"`
+	QueueMessages []QueueMessage `xml:"QueueMessage"`
+}
+
+// Used for modelling the xml we get back from Azure
+type QueueMessage struct {
+	XMLName         xml.Name        `xml:"QueueMessage"`
+	MessageId       string          `xml:"MessageId"`
+	InsertionTime   azureTimeFormat `xml:"InsertionTime"`
+	ExpirationTime  azureTimeFormat `xml:"ExpirationTime"`
+	DequeueCount    uint            `xml:"DequeueCount"`
+	PopReceipt      string          `xml:"PopReceipt"`
+	TimeNextVisible azureTimeFormat `xml:"TimeNextVisible"`
+	MessageText     string          `xml:"MessageText"`
+}
+
+// Used for modelling the json encoding of QueueMessage.MessageId that we get
+// back from Azure
+type TaskRun struct {
+	TaskId            string                  `json:"taskId"`
+	RunId             uint                    `json:"runId"`
+	QueueMessage      QueueMessage            `json:"-"`
+	SignedURLPair     SignedURLPair           `json:"-"`
+	ClaimCallSummary  queue.CallSummary       `json:"-"`
+	reclaimTimer      *time.Timer             `json:"-"`
+	TaskClaimRequest  queue.TaskClaimRequest  `json:"-"`
+	TaskClaimResponse queue.TaskClaimResponse `json:"-"`
+	Definition        queue.TaskDefinition1   `json:"-"`
+	Payload           GenericWorkerPayload    `json:"-"`
+}
```

Comment:
Having read everything else... I see that using one bug struct is convenient, though it forces you to remember which order things are constructed in.

To make it simpler, one could make a struct with sub-structs...  but the flow is fairly sane so I would just go with this... :)